hnn_core/lfp.py

```diff
@@ -0,0 +1,234 @@
+    seg_counts = list()
+    for sec in all_sections:
+        seg_counts.append(sec.nseg)
+    return seg_counts
+
+
+def _transfer_resistance(section, ele_pos, sigma, method):
+    """Transfer resistance between section and electrode position.
+
+    To arrive at the extracellular potential, the value returned by this
+    function is multiplied by the net transmembrane current flowing through all
+    segments of the section. Hence the term "resistance" (voltage equals
+    current times resistance).
+
+    Parameters
+    ----------
+    section : h.Section() The NEURON section. ele_pos : list (x, y, z) The x,
+        y, z coordinates of the electrode (in um) sigma : float Extracellular
+        conductivity (in S/m) method : str Approximation to use. 'psa' assigns
+        all transmembrane currents to the center point (0.5). 'lsa' treats the
+        section as a line source, but a single multiplier is calculated for
+        each section. Returns
+    -------
+    vres : list The resistance at each section.
+    """
+    ele_pos = np.array(ele_pos)  # electrode position to Numpy
+
+    start = np.array([section.x3d(0), section.y3d(0), section.z3d(0)])
+    end = np.array([section.x3d(1), section.y3d(1), section.z3d(1)])
+
+    if method == 'psa':
+
+        mid = (start + end) / 2.
+
+        # distance from section midpoint to electrode
+        dis = norm(ele_pos - mid)
+
+        # setting radius limit
+        if dis < section.diam / 2.0:
+            dis = section.diam / 2.0 + 0.1
+
+        phi = 1. / dis
+
+    elif method == 'lsa':
+        # From: Appendix C (pp. 137) in Holt, G. R. A critical reexamination of
+        # some assumptions and implications of cable theory in neurobiology.
+        # CalTech, PhD Thesis (1998).
+        #
+        #                      Electrode position
+        #   |------ L --------*
+        #                 b / | R
+        #                 /   |
+        #   0==== a ====1- H -+
+        #
+        # a: section orientation vector
+        # b: position vector of electrode with respect to section end (1)
+        # H: parallel distance from section end to electrode
+        # R: radial distance from section end to electrode
+        # L: parallel distance from section start to electrode
+        # Note that there are three distinct regimes to this approximation,
+        # depending on the electrode position along the section axis.
+        a = end - start
+        norm_a = norm(a)
+        b = ele_pos - end
+        # projection: H = a.cos(theta) = a.dot(b) / |a|
+        H = np.dot(b, a) / norm_a  # NB can be negative
+        L = H + norm_a
+        R2 = norm(b) ** 2 - H ** 2  # NB squares
+        # To avoid numerical errors when electrode is placed (anywhere) on the
+        # section axis, enforce minimal axial distance
+        R2 = max(R2, section.diam / 2.0 + 0.1)
+
+        if L < 0 and H < 0:  # electrode is "behind" section
+            num = np.sqrt(H ** 2 + R2) - H
```

Comment:
```suggestion
            num = b - H
```

would this make sense/simpler? same in the conditions below

Reply:
it would, but I hesitated because of how the equations are represented in papers. I'll simplify, but leave comments.